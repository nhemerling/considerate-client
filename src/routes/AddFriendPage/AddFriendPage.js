import React, { Component } from 'react';
import AddFriendForm from '../../components/AddFriendForm/AddFriendForm';
import FriendApiService from '../../services/friend-api-service';
import ConsiderateContext from '../../context/ConsiderateContext';

export default class AddFriendPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  static contextType = ConsiderateContext;

  handleAddFriendSuccess = (friend) => {
    FriendApiService.getFriends()
      .then((res) => {
        this.context.setFriendList(res);
        const { history } = this.props;
        history.push('/home');
      })
      .catch(this.context.setError);
  };

  render() {
    return (
      <section className="AddFriendPage">
        <h2>Add Friend</h2>
        <AddFriendForm onAddFriendSuccess={this.handleAddFriendSuccess} />
      </section>
    );
  }
}
