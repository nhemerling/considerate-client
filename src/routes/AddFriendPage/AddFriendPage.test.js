import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddFriendPage from './AddFriendPage';

describe('AddFriendPage component', () => {
  it('renders a AddFriendPage component', () => {
    const wrapper = shallow(<AddFriendPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
