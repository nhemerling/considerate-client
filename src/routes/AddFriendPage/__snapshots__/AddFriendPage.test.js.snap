// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFriendPage component renders a AddFriendPage component 1`] = `
<section
  className="AddFriendPage"
>
  <h2>
    Add Friend
  </h2>
  <AddFriendForm
    onAddFriendSuccess={[Function]}
  />
</section>
`;
